xwiki-commons-core/xwiki-commons-velocity/src/main/java/org/xwiki/velocity/tools/URLTool.java

```diff
@@ -58,4 +61,55 @@
         }
         return queryParams;
     }
+
+    /**
+     * Convert a map into a query string.
+     * 
+     * @param map the map to be converted
+     * @return the converted query string
+     * @since 10.4
+     */
+
+    public String getQueryString(Map<String, String> map)
+    {
+        String queryString = new String();
+        for (Map.Entry<String, String> entry : map.entrySet()) {
+            queryString = add(queryString, entry.getKey(), entry.getValue());
+        }
+
+        return queryString;
+    }
+
+    /**
+     * Add a parameter to a query string.
+     * 
+     * @param queryString the initial query string
+     * @param name the name of the parameter
+     * @param value the value of the parameter
+     * @return the new query string
+     * @since 10.4
+     */
+    public String add(String queryString, String name, String value)
```

Comment:
Since class is about URLs in general this method name should be more explicit on the fact that it's related to query strings.